src/ql/ir/compat/kernel.cc

```diff
@@ -1055,6 +1055,22 @@ void Kernel::gate(
     gates.extend(u.get_decomposition(qubits));
 }
 
+// adding state prepration / (arbitrary) qubit initialisation to kernel
+void Kernel::state_prep(
```

Comment:
It's something that is missing in the whole codebase, so not your fault, but this _should_ have an accompanying unit test. I have a PR in progress that adds the GoogleTest framework and some unit tests for passes. I intend to write those for kernel and other components as well.

Reply:
Fair and completely correct, I wish you success! And if you need me to write some tests for this code, or the general unitary decomposition, let me know, Id be happy to help.